msrestazure/polling/arm_polling.py

```diff
@@ -0,0 +1,417 @@
+                self.resource = None
+            else:
+                raise OperationFailed("Invalid status found")
+            return
+        raise OperationFailed("Operation failed or cancelled")
+
+    def get_status_from_location(self, response):
+        """Process the latest status update retrieved from a 'location'
+        header.
+
+        :param requests.Response response: latest REST call response.
+        :raises: BadResponse if response has no body and not status 202.
+        """
+        self._raise_if_bad_http_status_and_method(response)
+        code = response.status_code
+        if code == 202:
+            self.status = "InProgress"
+        else:
+            self.status = 'Succeeded'
+            if self._is_empty(response):
+                self.resource = None
+            else:
+                self.resource = self._deserialize(response)
+
+    def get_status_from_resource(self, response):
+        """Process the latest status update retrieved from the same URL as
+        the previous request.
+
+        :param requests.Response response: latest REST call response.
+        :raises: BadResponse if status not 200 or 204.
+        """
+        self._raise_if_bad_http_status_and_method(response)
+        if self._is_empty(response):
+            raise BadResponse('The response from long running operation '
+                              'does not contain a body.')
+
+        status = self._get_provisioning_state(response)
+        self.status = status or 'Succeeded'
+
+        self.resource = self._deserialize(response)
+
+    def get_status_from_async(self, response):
+        """Process the latest status update retrieved from a
+        'azure-asyncoperation' header.
+
+        :param requests.Response response: latest REST call response.
+        :raises: BadResponse if response has no body, or body does not
+         contain status.
+        """
+        self._raise_if_bad_http_status_and_method(response)
+        if self._is_empty(response):
+            raise BadResponse('The response from long running operation '
+                              'does not contain a body.')
+
+        self.status = self._get_async_status(response)
+        if not self.status:
+            raise BadResponse("No status found in body")
+
+    def set_async_url_if_present(self, response):
+        async_url = get_header_url(response, 'azure-asyncoperation')
+        if async_url:
+            self.async_url = async_url
+        location_url = get_header_url(response, 'location')
+        if location_url:
+            self.location_url = location_url
+
+    def get_status_link(self):
+        if self.async_url:
+            return self.async_url
+        elif self.location_url:
+            return self.location_url
+        elif self.method == "PUT":
+            return self.initial_response.request.url
+        else:
+            raise BadResponse("Unable to find a valid status link for polling")
+
+
+class ARMPolling(object):
```

Comment:
Shouldn't this inherit from msrest.polling.PollingMethod?

Reply:
Indeed!